subprojects/soak/src/integTest/groovy/org/gradle/vfs/VirtualFileSystemRetentionSoakTest.groovy

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.vfs
+
+import org.gradle.integtests.fixtures.daemon.DaemonIntegrationSpec
+import org.gradle.internal.os.OperatingSystem
+import org.gradle.internal.service.scopes.VirtualFileSystemServices
+import org.gradle.soak.categories.SoakTest
+import org.gradle.test.fixtures.file.TestFile
+import org.junit.experimental.categories.Category
+
+@Category(SoakTest)
+class VirtualFileSystemRetentionSoakTest extends DaemonIntegrationSpec {
+
+    private static final int NUMBER_OF_SUBPROJECTS = 50
+    private static final int NUMBER_OF_SOURCES_PER_SUBPROJECT = 100
+    private static final int TOTAL_NUMBER_OF_SOURCES = NUMBER_OF_SUBPROJECTS * NUMBER_OF_SOURCES_PER_SUBPROJECT
+    private static final double LOST_EVENTS_RATIO_MAC_OS = 0.75
+    private static final double LOST_EVENTS_RATIO_WINDOWS = 0.1
+
+    List<TestFile> sourceFiles
+
+    def setup() {
+        // Use 20 minutes idle timeout since the test may be running longer with an idle daemon
+        executer.withDaemonIdleTimeoutSecs(1200)
```

Comment:
Should the extra timeout be enabled only for the test with the idle daemon?

Reply:
Yeah, doing that now.